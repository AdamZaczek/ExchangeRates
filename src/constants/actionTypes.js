export const FETCH_CURRENCIES = 'FETCH_CURRENCIES';
export const ADD_CURRENCY = 'ADD_CURRENCY';
export const REMOVE_CURRENCY = 'FETCH_CURRENCY';
export const REMOVE_CURRENCIES = 'REMOVE_CURRENCIES';
