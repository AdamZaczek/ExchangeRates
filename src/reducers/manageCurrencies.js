//import { MANAGE_CURRENCIES } from '../constants/actionTypes';

const initialState = {
  currencies: [],
};

export default (state = initialState, action) => {
  switch (action.type) {
    case 'ADD_CURRENCY':
      if (state.currencies.includes(action.currency)) return state;
      return {
        ...state,
        currencies: [...state.currencies, action.currency],
      };
    default:
      return state;
  }
};
