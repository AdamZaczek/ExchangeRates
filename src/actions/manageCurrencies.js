import { ADD_CURRENCY, REMOVE_CURRENCY, REMOVE_CURRENCIES } from '../constants/actionTypes';

export const addCurrency = currency => ({
  type: ADD_CURRENCY,
  currency,
});

export const removeCurrency = currency => ({
  type: REMOVE_CURRENCY,
  currency,
});

export const removeAllCurrencies = currencies => ({
  type: REMOVE_CURRENCIES,
});
