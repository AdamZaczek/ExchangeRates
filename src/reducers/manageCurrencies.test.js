import manageCurrencies from './manageCurrencies';
import { ADD_CURRENCY, REMOVE_CURRENCY, REMOVE_CURRENCIES } from '../constants/actionTypes';

describe('manageCurrencies reducer', () => {
  test('should return default state', () => {
    expect(manageCurrencies(undefined, {})).toMatchSnapshot();
  });

  test('should return array without filtered element', () => {
    const c1 = {
      code: '1',
      currency: 'dollar'
    }
    const c2 = {
      code: '2',
      currency: 'euro'
    }
    const c3 = {
      code: '3',
      currency: 'frank'
    }
    const action = {
      type: REMOVE_CURRENCY,
      currency: c1,
    };
    expect(manageCurrencies({ currencies: [
      c1, c2, c3
    ] }, action)).toMatchSnapshot();
  });
});
