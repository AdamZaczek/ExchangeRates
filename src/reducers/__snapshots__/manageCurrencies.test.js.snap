// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manageCurrencies reducer should return array without filtered element 1`] = `
Object {
  "currencies": Array [
    Object {
      "code": "2",
      "currency": "euro",
    },
    Object {
      "code": "3",
      "currency": "frank",
    },
  ],
}
`;

exports[`manageCurrencies reducer should return default state 1`] = `
Object {
  "currencies": Array [],
}
`;
